addons/hr_payroll/static/src/js/benefit.js

```diff
@@ -0,0 +1,128 @@
+odoo.define('hr_payroll.benefit.view_custo', function(require) {
+    'use strict';
+
+    var core = require('web.core');
+    var CalendarController = require("web.CalendarController");
+
+    var _t = core._t;
+    CalendarController.include({
+
+        update: function () {
+            var self = this;
+            return this._super.apply(this, arguments).then(function () {
+                self._renderBenefitButtons();
+            });
+        },
+
+        _renderGenerateButton: function(date_from, date_to, employee_ids, secondary) {
+            var self = this;
+            var primary = !secondary ? 'btn-primary' : 'btn-secondary';
+            var txt = _t("Generate Benefits");
+            this.$buttons.find('.o_calendar_button_month').after(
+                $('<button class="btn ' + primary + ' btn-benefit" type="button">'+ txt +'</button>')
```

Comment:
This is an **unsafe** way of inserting text in a dom element. It may not be dangerous here but it's an anti-pattern that you must avoid at all costs. Use a proper QWeb template whenever possible, and when it's really overkill use the proper JQuery primitives such as addClass() and text() to avoid stupid XSS injections.

e.g.:
```var $button = $('<button class="btn" type="button">').addClass(primary).text(txt)```